src/ui/public/visualize/visualize_legend.js

```diff
@@ -96,11 +93,11 @@ uiModules.get('kibana')
       function refresh() {
         let vislibVis = $scope.renderbot.vislibVis;
 
-        if ($scope.uiState.get('vis.legendOpen') == null && vislibVis._attr.addLegend != null) {
-          $scope.open = vislibVis._attr.addLegend;
+        if ($scope.uiState.get('vis.legendOpen') == null && vislibVis.visConfigArgs.addLegend != null) {
```

Comment:
Is this why `visConfigArgs` is being exposed? These should be easy to convert to using a public api method like `vislibVis.hasLegend()` or something.